scene/resources/particle_process_material.cpp

```diff
@@ -166,23 +166,26 @@ void ParticleProcessMaterial::_update_shader() {
 		shader_map[mk].users++;
 		return;
 	}
-	//must create a shader!
+
+	// No pre-existing shader, create one.
 
 	// Add a comment to describe the shader origin (useful when converting to ShaderMaterial).
 	String code = "// NOTE: Shader automatically converted from " VERSION_NAME " " VERSION_FULL_CONFIG "'s ParticleProcessMaterial.\n\n";
 
 	code += "shader_type particles;\n";
 	code += "render_mode disable_velocity;\n";
-
 	if (collision_scale) {
 		code += "render_mode collision_use_scale;\n";
 	}
+	code += "\n";
+
+	// Define uniforms.
 
 	code += "uniform vec3 direction;\n";
 	code += "uniform float spread;\n";
 	code += "uniform float flatness;\n";
 
-	code += "uniform float inherit_emitter_velocity_ratio = 0;\n";
+	code += "uniform float inherit_emitter_velocity_ratio = 0.0;\n";
```

Comment:
I tried to make sure `float`s properly use `x.0` format, not `x` or `x.` or `.0`.